### ISSUE TITLE
Null pointer on my local adapter fields from the SectionedRecyclerViewAdapter constructor?

### ISSUE THREAD

**Issue by Xorok** [afollestad/sectioned-recyclerview#46]
- [X] I have verified there are no duplicate active or recent bugs, questions, or requests
- [X] I have verified that I am using the latest version of this library.
- [X] I have given my issue a non-generic title.

###### Include the following:
 - Library version: `0.4.4`
 - Device OS version: `7.1.1`
 - Device Manufacturer: `Android Emulator`
 - Device Name: `Nexus 5X base device`

Hello,
so I updated from v0.4.1 to v0.4.4 today, and now the app crashes with a Nullpointer from `getItemCount(int section)` where I call `entries.getSize()` on my adapter list. The list shouldn't be null though, as this is the constructor:

```java
CacheEntriesAdapter(CompositeUnmodifiableList<CacheEntry> entries) {
        Log.d(getClass().getSimpleName(), "DUDUDU");
        this.entries = entries;
}
```

I already checked several times, and the input list is definitely not null, it just doesn't seem to execute the constructor before calling `entries.getSize()`, as my Log.d's in the first line don't get printed before it crashes. This problem doesn't appear in v0.4.1, v0.4.2 or v0.4.3.

```java
06-07 13:42:48.478 19590-19590/io.github.madeinspengergasse.wischmopp D/AndroidRuntime: Shutting down VM
06-07 13:42:48.478 19590-19590/io.github.madeinspengergasse.wischmopp E/AndroidRuntime: FATAL EXCEPTION: main
                                                                                        Process: io.github.madeinspengergasse.wischmopp, PID: 19590
                                                                                        java.lang.NullPointerException: Attempt to invoke virtual method 'int io.github.madeinspengergasse.wischmopp.util.CompositeUnmodifiableList.sizeOfList1()' on a null object reference
                                                                                            at io.github.madeinspengergasse.wischmopp.scan.view.CacheEntriesAdapter.getItemCount(CacheEntriesAdapter.java:43)
                                                                                            at com.afollestad.sectionedrecyclerview.PositionManager.invalidate(PositionManager.java:23)
                                                                                            at com.afollestad.sectionedrecyclerview.SectionedRecyclerViewAdapter.<init>(SectionedRecyclerViewAdapter.java:32)
                                                                                            at io.github.madeinspengergasse.wischmopp.scan.view.CacheEntriesAdapter.<init>(CacheEntriesAdapter.java:28)
                                                                                            at io.github.madeinspengergasse.wischmopp.scan.view.ScanFragment.showData(ScanFragment.java:142)
                                                                                            at io.github.madeinspengergasse.wischmopp.scan.presenter.ScanPresenter$1.lambda$onGetStatsCompleted$0$ScanPresenter$1(ScanPresenter.java:120)
                                                                                            at io.github.madeinspengergasse.wischmopp.scan.presenter.ScanPresenter$1$$Lambda$0.run(Unknown Source)
                                                                                            at android.os.Handler.handleCallback(Handler.java:751)
                                                                                            at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                                            at android.os.Looper.loop(Looper.java:154)
                                                                                            at android.app.ActivityThread.main(ActivityThread.java:6119)
                                                                                            at java.lang.reflect.Method.invoke(Native Method)
                                                                                            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
                                                                                            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```
 
###### Reproduction Steps

1. Create an args constructor in SectionedRecyclerViewAdapter in which you pass your entries list
2. Run

You can find the current, non-working state of our app with this problem here: https://github.com/Xorok/Wischmopp/tree/testing/adrian
The important classes are CacheEntriesAdapter, ScanPresenter & ScanFragment in the scan package. You just have to press the big red button on the first screen to see the problem in action. If you decide to take a look at the project, any comments or possible improvements are welcome 😁

###### Expected Result
It doesn't call `entries.getSize()` before executing the constructor, the list gets initialized and the Nullpointer doesn't happen.

###### Actual Result
The app crashes with a Nullpointer.

**Comment by afollestad** [comment]
Well, you need to call `super()` in the first line of your constructor.

**Comment by Xorok** [comment]
Still doesn't work :/

**Comment by afollestad** [comment]
You tried that in https://github.com/Xorok/Wischmopp/blob/testing/adrian/app/src/main/java/io/github/madeinspengergasse/wischmopp/scan/view/CacheEntriesAdapter.java?

The reason it's crashing is because `getItemCount(int section)` gets called immediately from `SectionedRecyclerViewAdapter`'s constructor through `positionManager.invalidate(this)`:

```java
public SectionedRecyclerViewAdapter() {
    positionManager = new PositionManager();
    positionManager.invalidate(this);
}
```

**Comment by afollestad** [comment]
Oh, wait, that's the reason it's crashing. Your `getItemCount(int section)` method is accessing your lists before they're instantiated. Try instantiating your lists *before* calling `super()`.

**Comment by Xorok** [comment]
It doesn't compile if I switch the order of the two statements.

> Call to 'super()' must be first statement in constructor body

**Comment by afollestad** [comment]
Ah, right.

So I thought about it a bit more, the safest thing for you do would be to check whether or not your lists are null in `getItemCount(int section)`. If they are, return 0.

**Comment by Xorok** [comment]
That'll do it for the moment, but this is more of a workaround than a real solution.
Passing your data to the adapter through an args constructor is common practice and should work like it does with the regular adapter. It is even done so in the official docs: https://developer.android.com/training/material/lists-cards.html

**Comment by afollestad** [comment]
That is a good point, I'll see what I can do to improve the internal logic.

**Comment by afollestad** [comment]
Fixed in 0.5.0
